packages/editor-ui/src/components/mixins/externalHooks.ts

```diff
@@ -0,0 +1,38 @@
+import Vue from "vue";
+import { IExternalHooks } from "@/Interface";
+import { IDataObject } from "n8n-workflow";
+
+export async function runExternalHook(
+	eventName: string,
+	// tslint:disable-next-line: no-any
+	state: any,
+	metadata?: IDataObject,
+) {
+	// @ts-ignore
+	if (!window.externalHooks) {
+		return;
+	}
+
+	const [resource, operator] = eventName.split(".");
+
+	// @ts-ignore
+	if (window.externalHooks[resource] && window.externalHooks[resource][operator]) {
+		// @ts-ignore
+		const hookMethods = window.externalHooks[resource][operator];
+		for (const hookmethod of hookMethods) {
+			hookmethod(state, metadata);
```

Comment:
We should probably have an "await" here.

Reply:
was going for no await approach here, changing now